content/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1.md

```diff
@@ -0,0 +1,161 @@
+
+```json
+{
+...
+  "subject": {
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "digest": "sha256:5b0bca...",
+    "size": 7682
+  },
+...
+}
+```
+
+This is the primary mechanism for linking objects to one another. This
+is what is used to determine the proper list returned by the registry
+API endpoint described in #3 below.
+
+If the registry supports the processing of the `subject` field, it is required
+to respond with a header in the following form: `OCI-Subject: sha256:5b0bca...`
+(the value being the `digest` field in the `subject` JSON object).
+
+## 3. New Registry API Endpoint for Querying Relationships
+
+The following HTTP API endpoint has been added to enable
+querying for relationships between objects in the registry:
+`GET /v2/<name>/referrers/<digest>`
+
+The contents of a valid response to this request is a valid OCI
+image index containing a list of descriptors which point to the
+provided digest:
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 1234,
+      "digest": "sha256:a1a1a1...",
+      "artifactType": "application/vnd.example.sbom.v1",
+      "annotations": {
+        "org.opencontainers.artifact.created": "2022-01-01T14:42:55Z",
+        "org.example.sbom.format": "json"
+      }
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 1234,
+      "digest": "sha256:a2a2a2...",
+      "artifactType": "application/vnd.example.signature.v1",
+      "annotations": {
+        "org.opencontainers.artifact.created": "2022-01-01T07:21:33Z",
+        "org.example.signature.fingerprint": "abcd"
+      }
+    }
+  ]
+}
+```
+
+If the response contains a `Link` header
+(see [RFC5988](https://www.rfc-editor.org/rfc/rfc5988.html)),
+this indicates that there is another page of results which the client
+should fetch.
+
+In addition, clients may provide a filter in querystring in the form
+`?artifactType=<mediaType>`. Registries may or may not support this
+server-side filtering mechanism. If supported, however, the registry is
+required to respond with a header indicating the filter(s) applied:
+`OCI-Filters-Applied: artifactType`.
+
+Note: the only filter defined at this time is `artifactType`.
+
+# Closing Thoughts
+
+This post attemts to cover all of the major changes involved in adopting
+OCI Image and Distribution specs v1.1, however there are several other
+fine-grained details which can only be understood by carefully reading
+through the specifications.
+
+Some other notable changes include the following:
+
+- `data` field in the descriptor
+- non-distributable layers are deprecated
+- zstd compression support
+- extension support in distribution-spec
+- support for resuming chunked blob push
+- anonymous blob mount support in distribution-spec
+
```

Comment:
https://github.com/opencontainers/distribution-spec/pull/393 might be a good one to add here ("a `Warning` header for surfacing user-facing client warnings" or something) :eyes:

Reply:
good call - done